lib/chef/provider/package/zypper.rb

```diff
@@ -70,7 +71,34 @@ def get_current_versions
         end
 
         def candidate_version
-          @candidate_version ||= package_name_array.each_with_index.map { |pkg, i| available_version(i) }
+          if source_files_exist?
```

Comment:
This is presuming the source property is a string, which it is for the `package` resource, but for the `dpkg_package` resource that you're modeling after it can be an array.